s3-endpoints.go

```diff
@@ -36,6 +37,7 @@ var awsS3EndpointMap = map[string]string{
 	"sa-east-1":      "s3-sa-east-1.amazonaws.com",
 	"us-gov-west-1":  "s3-us-gov-west-1.amazonaws.com",
 	"cn-north-1":     "s3.cn-north-1.amazonaws.com.cn",
+	"cn-northwest-1": "s3.cn-northwest-1.amazonaws.com.cn",
```

Comment:
It looks like the following codes in `api.go` and `utilities.go` files also need to be touched to accommodate the newly added region.

api.go
```
func getRegionFromURL(u url.URL) (region string) {
	region = ""
	if s3utils.IsGoogleEndpoint(u) {
		return
	} else if s3utils.IsAmazonChinaEndpoint(u) {
		// For china specifically we need to set everything to
		// cn-north-1 for now, there is no easier way until AWS S3
		// provides a cleaner compatible API across "us-east-1" and
		// China region.
		return "cn-north-1"
	} else if s3utils.IsAmazonGovCloudEndpoint(u) {
        :
        :
```

utilities.go
```
func getDefaultLocation(u url.URL, regionOverride string) (location string) {
	if regionOverride != "" {
		return regionOverride
	}
	if s3utils.IsAmazonChinaEndpoint(u) {
		return "cn-north-1"
	}
        :
        :
```